pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -222,27 +226,42 @@ public void build()
       TransformPipeline.Result reusedResult = new TransformPipeline.Result();
       while (_recordReader.hasNext()) {
         long recordReadStartTime = System.currentTimeMillis();
-        long recordReadStopTime;
+        long recordReadStopTime = System.currentTimeMillis();
         long indexStopTime;
         reuse.clear();
-        GenericRow decodedRow = _recordReader.next(reuse);
-        recordReadStartTime = System.currentTimeMillis();
-        _transformPipeline.processRow(decodedRow, reusedResult);
-        recordReadStopTime = System.currentTimeMillis();
-        _totalRecordReadTime += (recordReadStopTime - recordReadStartTime);
+        try {
+          GenericRow decodedRow = _recordReader.next(reuse);
+          recordReadStartTime = System.currentTimeMillis();
+          _transformPipeline.processRow(decodedRow, reusedResult);
+          recordReadStopTime = System.currentTimeMillis();
+          _totalRecordReadTime += (recordReadStopTime - recordReadStartTime);
+        } catch (Exception e) {
+          if (_continueOnError) {
+            incompleteRowsFound++;
+            LOGGER.debug("Error occurred while reading row during indexing", e);
+            continue;
+          }
+        }
```

Comment:
Do we have documentation on what "Continue on Error" flag mean? I thought if it's set to false or not set, we want to stop the ingestion. If that's true, then we need to rethrow if _continueOnError is not set?

Reply:
Yes, you are correct. If it is not set then we need to re-throw the error and stop ingestion. Let me fix it here.